spec/rubocop/cop/rspec/sort_metadata_spec.rb

```diff
@@ -0,0 +1,217 @@
+      end
+    RUBY
+
+    expect_correction(<<~RUBY)
+      context 'Something', baz: true, foo: 'bar' do
+      end
+    RUBY
+  end
+
+  it 'does not register an offense when using mixed metadata ' \
+     'in alphabetical order (respectively)' do
+    expect_no_offenses(<<~RUBY)
+      it 'Something', :a, :b, baz: true, foo: 'bar' do
+      end
+    RUBY
+  end
+
+  it 'registers an offense when using mixed metadata, ' \
+     'but only the hash keys are in alphabetical order' do
+    expect_offense(<<~RUBY)
+      it 'Something', :b, :a, baz: true, foo: 'bar' do
+                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Sort metadata alphabetically.
+      end
+    RUBY
+
+    expect_correction(<<~RUBY)
+      it 'Something', :a, :b, baz: true, foo: 'bar' do
+      end
+    RUBY
+  end
+
+  it 'registers an offense when using mixed metadata, ' \
+     'but only the symbol keys are in alphabetical order' do
+    expect_offense(<<~RUBY)
+      it 'Something', :a, :b, foo: 'bar', baz: true do
+                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Sort metadata alphabetically.
+      end
+    RUBY
+
+    expect_correction(<<~RUBY)
+      it 'Something', :a, :b, baz: true, foo: 'bar' do
+      end
+    RUBY
+  end
+
+  it 'registers an offense when using mixed metadata ' \
+     'and both symbols metadata and hash keys are not in alphabetical order' do
+    expect_offense(<<~RUBY)
+      it 'Something', :b, :a, foo: 'bar', baz: true do
+                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Sort metadata alphabetically.
+      end
+    RUBY
+
+    expect_correction(<<~RUBY)
+      it 'Something', :a, :b, baz: true, foo: 'bar' do
+      end
+    RUBY
+  end
+
+  it 'registers an offense when using mixed metadata ' \
+     'and both symbols metadata and hash keys are not in alphabetical order ' \
+     'and the hash values are complex objects' do
+    expect_offense(<<~RUBY)
+      it 'Something', variable, 'B', :a, key => {}, foo: ->(x) { bar(x) }, Identifier.sample => true, baz: Snafu.new do
+                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Sort metadata alphabetically.
+      end
+    RUBY
+
+    expect_correction(<<~RUBY)
+      it 'Something', :a, 'B', variable, baz: Snafu.new, foo: ->(x) { bar(x) }, Identifier.sample => true, key => {} do
+      end
+    RUBY
+  end
+
+  it 'registers an offense only when example or group has a block' do
+    expect_offense(<<~RUBY)
+      shared_examples 'a difficult situation', 'B', :a do |x, y|
+                                               ^^^^^^^ Sort metadata alphabetically.
+      end
+
+      include_examples 'a difficult situation', 'value', 'another value'
+    RUBY
+
+    expect_correction(<<~RUBY)
+      shared_examples 'a difficult situation', :a, 'B' do |x, y|
+      end
+
+      include_examples 'a difficult situation', 'value', 'another value'
+    RUBY
+  end
+
+  it 'does not register an offense ' \
```

Comment:
I wonder why is that? It still would be nice to have it sorted

Reply:
I agree, we should flag multiline offenses too. I don’t know if we should attempt to autocorrect them though.

Reply:
In the multi line case, when I provide the range, it will only put carets under the first line of the metadata, even though the range goes further. Is that intentional or am I doing something wrong?